systemtest/src/test/java/io/strimzi/systemtest/upgrade/kraft/KRaftKafkaUpgradeDowngradeST.java

```diff
@@ -0,0 +1,328 @@
+                    .editKafka()
+                        .withVersion(initialVersion.version())
+                        .withConfig(null)
+                    .endKafka()
+                .endSpec();
+
+            // Do not set metadataVersion if it's not passed to method
+            if (initMetadataVersion != null) {
+                kafka
+                    .editSpec()
+                        .editKafka()
+                            .withMetadataVersion(initMetadataVersion)
+                        .endKafka()
+                    .endSpec();
+            }
+
+            resourceManager.createResourceWithWait(testContext,
+                KafkaNodePoolTemplates.kafkaNodePoolWithControllerRoleAndPersistentStorage(TestConstants.CO_NAMESPACE, CONTROLLER_NODE_NAME, clusterName, controllerReplicas).build(),
+                KafkaNodePoolTemplates.kafkaNodePoolWithBrokerRoleAndPersistentStorage(TestConstants.CO_NAMESPACE, BROKER_NODE_NAME, clusterName, brokerReplicas).build(),
+                kafka.build()
+            );
+
+            // ##############################
+            // Attach clients which will continuously produce/consume messages to/from Kafka brokers during rolling update
+            // ##############################
+            // Setup topic, which has 3 replicas and 2 min.isr to see if producer will be able to work during rolling update
+            resourceManager.createResourceWithWait(testContext, KafkaTopicTemplates.topic(clusterName, continuousTopicName, 3, 3, 2, TestConstants.CO_NAMESPACE).build());
+            String producerAdditionConfiguration = "delivery.timeout.ms=20000\nrequest.timeout.ms=20000";
+
+            KafkaClients kafkaBasicClientJob = new KafkaClientsBuilder()
+                .withProducerName(producerName)
+                .withConsumerName(consumerName)
+                .withBootstrapAddress(KafkaResources.plainBootstrapAddress(clusterName))
+                .withTopicName(continuousTopicName)
+                .withMessageCount(continuousClientsMessageCount)
+                .withAdditionalConfig(producerAdditionConfiguration)
+                .withDelayMs(1000)
+                .build();
+
+            resourceManager.createResourceWithWait(testContext, kafkaBasicClientJob.producerStrimzi());
+            resourceManager.createResourceWithWait(testContext, kafkaBasicClientJob.consumerStrimzi());
+            // ##############################
+
+        } else {
+            LOGGER.info("Initial Kafka version (" + initialVersion.version() + ") is already ready");
+            controllerPods = PodUtils.podSnapshot(TestConstants.CO_NAMESPACE, controllerSelector);
+            brokerPods = PodUtils.podSnapshot(TestConstants.CO_NAMESPACE, brokerSelector);
+
+            // Wait for metadataVersion change
+            if (!sameMinorVersion
+                && !isUpgrade
+                && !testContext.getDisplayName().contains("DowngradeToOlderMetadataVersion")) {
```

Comment:
What does this mean?

Reply:
The `!testContext.getDisplayName().contains("DowngradeToOlderMetadataVersion")` check is now removed as I removed the test that it mentions.

Otherwise I guess this if block is not relevant, as it should change the `metadataVersion` in case that we are doing downgrade test and we have different minor version (so f.e. downgrade from 3.6 -> 3.5) -> that's not possible.